internal/controller/topic/topic.go

```diff
@@ -184,10 +184,47 @@ func (c *external) Create(ctx context.Context, mg resource.Managed) (managed.Ext
 	return managed.ExternalCreation{}, nil
 }
 
-func (c *external) Update(_ context.Context, _ resource.Managed) (managed.ExternalUpdate, error) {
-	c.log.Info("topic updates not supported yet!")
-	// todo(turkenh): Support topic updates
-	return managed.ExternalUpdate{}, nil
+func (c *external) Update(ctx context.Context, mg resource.Managed) (managed.ExternalUpdate, error) {
+	cr, ok := mg.(*v1alpha1.Topic)
+	if !ok {
+		return managed.ExternalUpdate{}, errors.New(errNotTopic)
+	}
+
+	td, err := c.kafkaClient.ListTopics(ctx, meta.GetExternalName(cr))
+	if err != nil {
+		return managed.ExternalUpdate{}, err
+	}
+
+	p, ok := td[meta.GetExternalName(cr)]
+	if !ok || errors.Is(p.Err, kerr.UnknownTopicOrPartition) {
+		return managed.ExternalUpdate{}, nil
+	}
+	if p.Err != nil {
+		return managed.ExternalUpdate{}, errors.Wrapf(p.Err, "cannot get topic")
+	}
+
+	l := cr.Spec.ForProvider.Partitions - len(p.Partitions)
+
+	if l < 1 {
+		return managed.ExternalUpdate{}, errors.New("cannot decrease partition count")
+	} else {
+		resp, err := c.kafkaClient.UpdatePartitions(ctx, cr.Spec.ForProvider.Partitions, meta.GetExternalName(cr))
+
```

Comment:
```suggestion
	if l < 1 {
		return managed.ExternalUpdate{}, errors.New("cannot decrease partition count")
	} 
	resp, err := c.kafkaClient.UpdatePartitions(ctx, cr.Spec.ForProvider.Partitions, meta.GetExternalName(cr))
```
`else` could be omitted and early returns in if conditions are idiomatic in go: https://golang.org/doc/effective_go#if